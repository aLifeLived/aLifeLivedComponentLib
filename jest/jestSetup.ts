/* eslint-disable no-undef */
jest.useFakeTimers({ legacyFakeTimers: true }); // Stops reference error from appearing
