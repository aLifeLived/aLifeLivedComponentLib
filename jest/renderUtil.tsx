import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';
import { library } from '@fortawesome/fontawesome-svg-core';
import { iconList } from '../src/themes/systemIcon/IconList';

library.add(iconList);

function render(ui: any, { ...renderOptions } = {}) {
  function Wrapper({ children }: { children: any }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
