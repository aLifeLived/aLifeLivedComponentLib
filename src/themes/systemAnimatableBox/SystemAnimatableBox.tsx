import {
  backgroundColor,
  BackgroundColorProps,
  border,
  BorderProps,
  BoxProps,
  color,
  ColorProps,
  createRestyleComponent,
  layout,
  LayoutProps,
  spacing,
  SpacingProps,
} from '@shopify/restyle';
import React from 'react';
import { View, AnimatableProperties } from 'react-native-animatable';
import { Theme } from '../theme';

export type StyledSystemAnimatableProps = AnimatableProperties<Theme> &
  SpacingProps<Theme> &
  LayoutProps<Theme> &
  BorderProps<Theme> &
  BackgroundColorProps<Theme> &
  BoxProps<Theme> &
  ColorProps<Theme> & {
    children?: React.ReactNode;
  };

export const SystemAnimatableBox = createRestyleComponent<
  StyledSystemAnimatableProps,
  Theme
>([spacing, layout, border, backgroundColor, color], View);
