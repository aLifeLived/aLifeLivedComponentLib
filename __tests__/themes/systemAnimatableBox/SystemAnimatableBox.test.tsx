import React from 'react';
import { render } from '../../../jest/renderUtil';

// Component
import { SystemAnimatableBox } from '../../../src/themes/systemAnimatableBox/SystemAnimatableBox';

const setup = () => {
  const utils = render(<SystemAnimatableBox />);

  return {
    ...utils,
  };
};

describe('themes >> systemAnimatableBox >> SystemAnimatableBox', () => {
  test('Should match snapshot', () => {
    const { toJSON } = setup();
    expect(toJSON()).toMatchSnapshot();
  });
});
