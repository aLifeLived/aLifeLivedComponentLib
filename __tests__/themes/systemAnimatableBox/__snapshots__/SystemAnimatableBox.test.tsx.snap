// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themes >> systemAnimatableBox >> SystemAnimatableBox Should match snapshot 1`] = `
<View
  forwardedRef={[Function]}
  style={Object {}}
/>
`;
